engine/src/main/java/io/zeebe/engine/processor/StreamProcessorController.java

```diff
@@ -250,11 +250,17 @@ private void resume() {
   }
 
   public ActorFuture<Long> getLastProcessedPositionAsync() {
-    return processingStateMachine.getLastProcessedPositionAsync();
+    final CompletableActorFuture<Long> future = new CompletableActorFuture();
+    actor.call(
+        () -> future.complete(processingStateMachine.getLastSuccessfulProcessedEventPosition()));
+    return future;
   }
 
   public ActorFuture<Long> getLastWrittenPositionAsync() {
-    return processingStateMachine.getLastWrittenPositionAsync();
+    final CompletableActorFuture<Long> future = new CompletableActorFuture();
+    actor.call(
```

Comment:
Actor.call returns a future with the callable return value, so you can use this instead of a separate future